### DISCUSSION TITLE
Force group names even for single dependency

### DISCUSSION THREAD

**Discussion by TWiStErRob** [renovatebot/renovate#25588] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want the single-dependency updates like
 * https://github.com/TWiStErRob/net.twisterrob.cinema/pull/113
 * https://github.com/TWiStErRob/net.twisterrob.cinema/pull/171

to be titled the same way as many-dependency-in-a-group updates
 * https://github.com/TWiStErRob/net.twisterrob.cinema/pull/584

---

Note: I have a custom commit message, you can ignore that for this issue:
```json
"commitMessageExtra": "from {{{currentValue}}} to {{{newValue}}}{{#if isMajor}} (major v{{{newMajor}}}){{else}}{{/if}}",
```
https://github.com/TWiStErRob/renovate-config/blob/2587a12aa40310f6aea9bdb40a281436458f086b/default.json#L10

---

**The key here is that I always want the Monorepo's group name in the PR title, even if it's just one dependency being updated.**

### What you tried so far.

I have a group set up for "Kotlin":
```json
"packageRules": [
	{
		"groupName": "Kotlin",
		"description": "Group together all dependencies from Kotlin, Dokka and KotlinX, including coroutines.",
		"matchPackagePatterns": [
			"^org\\.jetbrains\\.kotlin:kotlin-.*$",
			"^org\\.jetbrains\\.kotlin\\.(.+):org\\.jetbrains\\.kotlin\\.(\\$1)\\.gradle\\.plugin$"
		],
```
https://github.com/TWiStErRob/renovate-config/blob/2587a12aa40310f6aea9bdb40a281436458f086b/group.json#L34-L42

---

This was set up even at the time of the other updates! However, since maven publishing might happen slowly on the Kotlin side, or some dependencies have slightly different cadence, Renovate picked up and merged some singular updates. I've since added `"minimumReleaseAge": "8 hours"` to reduce the possibility of this, but I'm still getting occasional single-update PRs. Especially in repos where there is only one dependency. **I would like to have this work consistently across all repos regardless of number references from a group.**

### Relevant debug logs

Not sure if there are relevant logs.

**Comment by rarkins** [comment]
This is not a first-class config option but could potentially be "hacked" as you're trying.

I originally had this configuration option but removed it years ago after nobody seemed to use it. I referred to it as something like "lazy" or "eager" group PR title naming (with lazy being the default like today).

We could make it a boolean option now, something like `groupNamePrTitleAlways`, defaulting to false. You're welcome to submit a PR for this, if you wish

**Comment by NikolayFrantsev** [comment]
It’s not very convenient, but the requested behavior can be achieved by overriding the commit message properties:

```
  "packageRules": [{
      "matchPackageNames": ["/.*eslint.*/"],
      "groupName": "eslint with plugins",

      "commitMessageTopic": "eslint with plugins",
      "commitMessageExtra": ""
    }]
```

**Reply by TWiStErRob** [reply]
I explicitly have `commitMessageExtra` overridden, is setting it to empty something required for this to work, or you think it's irrelevant?

**Reply by NikolayFrantsev** [reply, marked as answer]
For your specific case, I believe you should set "commitMessageTopic" to "Kotlin" and keep "commitMessageExtra" with your custom value. That will force single-group commits to be named the same way as commits with multiple packages.

**Reply by TWiStErRob** [reply]
Confirmed in https://github.com/TWiStErRob/glide-support/commit/98688a4efba9706ab2f754e85850a457addda002
It changed https://github.com/TWiStErRob/glide-support/pull/481 as expected.

**Reply by TWiStErRob** [reply]
For the record I also tried [adding a global](https://github.com/TWiStErRob/renovate-config/pull/87)
```json5
commitMessageTopic: "{{#if groupName}}{{{groupName}}}{{else}}{{{depName}}}{{/if}}"
```
but it did this:
```patch
-Update dependency com.android.application from 9.2.0-rc01 to 9.3.0-alpha01
+Update com.android.application from 9.2.0-rc01 to 9.3.0-alpha01
```

and I tried a [package rule where I set the group](https://github.com/TWiStErRob/renovate-config/pull/88):
```json5
groupName: "Android Gradle Plugin",
commitMessageTopic: "{{{groupName}}}",
```
and that resulted in
```patch
-Update dependency com.android.application from 9.2.0-rc01 to 9.3.0-alpha01
+Update from 9.2.0-rc01 to 9.3.0-alpha01
```

so apparently only the [hard override](https://github.com/TWiStErRob/renovate-config/pull/89) [works](https://github.com/TWiStErRob/glide-support/pull/481).

**Comment by felipecrs** [comment]
This should handle it more gracefully:

- https://github.com/renovatebot/renovate/pull/43629

Example:

```diff
"packageRules": [
	{
		"groupName": "Kotlin",
		"description": "Group together all dependencies from Kotlin, Dokka and KotlinX, including coroutines.",
		"matchPackagePatterns": [
			"^org\\.jetbrains\\.kotlin:kotlin-.*$",
			"^org\\.jetbrains\\.kotlin\\.(.+):org\\.jetbrains\\.kotlin\\.(\\$1)\\.gradle\\.plugin$"
		],
+		"groupSingleUpdates": true
	}
]
```
